### PR TITLE
tests: add comments for clientv3test.TestWatchResumeInitRev

### LINKED ISSUE #16867: Improve comments for clientv3test.TestWatchResumeInitRev
### What would you like to be added?

Please review the test and provide summary at the beginning of the test.

### Why is this needed?

improving readability

### PR DESCRIPTION
**Pull request by greenmoon55**
Fixes #16867

### CODE REVIEW

File: tests/integration/clientv3/watch_test.go [hunk 1]
```diff
@@ -350,6 +350,8 @@ func putAndWatch(t *testing.T, wctx *watchctx, key, val string) {
 	}
 }
 
+// TestWatchResumeInitRev tests watch resume with disconnect before the first event.
+// It ensures that correct events are returned corresponding to the start revision. 
 func TestWatchResumeInitRev(t *testing.T) {
```
Comment (lavacat) [inline tests/integration/clientv3/watch_test.go:355]: Let's rename the test to `TestWatchResumeAfterDisconnect`

File: tests/integration/clientv3/watch_test.go [hunk 2]
```diff
@@ -392,6 +397,7 @@ func TestWatchResumeInitRev(t *testing.T) {
 		if len(resp.Events) == 0 {
 			t.Fatal("expected event on watch")
 		}
+		// Events should be put(a, 3) and put(a, 4)
```
Comment (lavacat) [inline tests/integration/clientv3/watch_test.go:400]: nit: can you also add verification for (a, 4) and then we got only 2 events?

**Review by lavacat — COMMENTED** [review]
Looks good, some small comments.

**Comment by greenmoon55** [comment]
@lavacat I updated the pr and fixed the `go fmt` error.

**Review by lavacat — APPROVED** [review]

**Comment by greenmoon55** [comment]
Do I need another approval?

**Review by jmhbnz — APPROVED** [review]
LGTM - Thanks @greenmoon55 

Gentle ping @ahrtr for merge.

**Review by ahrtr — APPROVED** [review]

**Pull request merged**